horace_core/sqw/coord_transform/@ubmat_proj/ubmat_proj.m

```diff
@@ -0,0 +1,387 @@
+classdef ubmat_proj < line_proj_interface
+    %  Class defines coordinate transformations necessary to support legacy
+    %  Horace cuts in crystal coordinate system (orthogonal or non-orthogonal)
+    %
+    %  Defines coordinate transformations, used by cut_sqw when making
+    %  Horace cuts defined by rotation matrix
+    %
+    %  Object that defines the ortholinear projection operations
+    %
+    % Input accepting the structure:
+    %   >> proj = ubmat_proj(proj_struct)
+    %             where proj_struct is the
+    %             structure, containing any fields, with names, equal any
+    %             public fields of the line_proj class.
+    %
+    % As a standard serializable class, class line_proj accepts full set of
+    % positional and key-value parameters, which constitute its properties
+    %
+    % Argument input:
+    %   >> proj = ubmat_proj(u_to_rlu)
+    %   >> proj = ubmat_proj(u_to_rlu,scale)
+    %
+    %   Full positional arguments input (can be truncated at any argument
+    %   leaving other arguments default):
+    %   >> proj = ubmat_proj(u_to_rlu,scale,alatt,angdeg,offset,...
+    %                        label,title,lab1,lab2,lab3,lab4)
+    %
+    %   plus any of other arguments, provided as key-value pair e.g.:
+    %
+    %   >> proj = ubmat_proj(...,'offset',offset,...)
+    %   >> proj = ubmat_proj(...,'label',labelcellstr,...)
+    %   >> proj = ubmat_proj(...,'lab1',labelstr,...)
+    %                   :
+    %   >> proj = ubmat_proj(...,'lab4',labelstr,...)
+    %
+    % Minimal fully functional form:
+    %   >> proj =  ubmat_proj(u_to_rlu,'alatt',latice_parameters,'angdeg',lattice_angles_in_degrees);
```

Comment:
It says above that you can truncate at any argument. How is this, then, the minimal fully functional form?

Reply:
Because if you have not provide these three arguments you can not use it. It will need more parameters

Description modified